dashboard/test/ui/features/hourOfCode_signedIn.feature

```diff
@@ -61,16 +61,22 @@ Scenario: Failing at puzzle 6, refreshing puzzle 6, bubble should show up as att
 
 Scenario: Async progress write followed by a stale read
   Given I am on "http://studio.code.org/hoc/20?noautoplay=true"
-  And element ".header_middle a.level_link:first" does not have class "perfect"
+  And I wait to see ".header_middle"
+  And element ".header_middle a.level_link:last" does not have class "perfect"
   Then mark the current level as completed on the client
-  And I reload the page
+  And validate progress for level is "perfect"
+
+Scenario: Progress on the server that is not on the client
```

Comment:
Is this scenario name a precondition or a desired outcome? I can't tell.  Maybe rephrase to be more explicit about the expected behavior.